src/Illuminate/View/Compilers/Concerns/CompilesWhens.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Illuminate\View\Compilers\Concerns;
+
+trait CompilesWhens
+{
+    /**
+     * Compile the 'when' statement into valid PHP.
+     *
+     * @param  string  $expression
+     * @param  string  $output
+     * @param  bool  $escape
+     * @return string
+     */
+    protected function compileWhen($expression)
```

Comment:
I would move this to [src/Illuminate/View/Compilers/Concerns/CompilesConditionals.php](https://github.com/laravel/framework/blob/11.x/src/Illuminate/View/Compilers/Concerns/CompilesConditionals.php) instead of creating a new trait.